sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -129,11 +143,28 @@ export const imdsMsi: MSI = {
       `Using the IMDS endpoint coming form the environment variable MSI_ENDPOINT=${process.env.MSI_ENDPOINT}, and using the cloud shell to proceed with the authentication.`
     );
 
-    return msiGenericGetToken(
-      identityClient,
-      prepareRequestOptions(resource, clientId),
-      expiresInParser,
-      getTokenOptions
+    for (let retries = 0; retries < imdsMsiRetryConfig.maxRetries; retries++) {
+      try {
+        return await msiGenericGetToken(
+          identityClient,
+          prepareRequestOptions(resource, clientId),
+          expiresInParser,
+          getTokenOptions
+        );
+      } catch (error) {
+        if (error.statusCode === 404) {
+          const nextDelay = imdsMsiRetryConfig.nextDelayInMs;
+          imdsMsiRetryConfig.nextDelayInMs *= imdsMsiRetryConfig.intervalIncrement;
```

Comment:
since `imdsMsiRetryConfig` is an object in module scope, this will affect other calls to getToken, right?

Maybe `imdsMsiRetryConfig.nextDelayInMs` could be renamed to `initialDelayInMs` and then be assigned to a more locally scoped variable so this doesn't happen?

Reply:
ohhh sorry! Thank you so much! I had this originally in the local scope. My bad

Reply:
Thank you!

Reply:
I accidentally picked the name `startDelayInMs` instead, but otherwise I've aligned to what you requested! Let me know if the new code is ok 👍